backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -0,0 +1,207 @@
+package org.ovirt.engine.core.bll.scheduling.utils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.ovirt.engine.core.common.businessentities.CpuPinningPolicy;
+import org.ovirt.engine.core.common.businessentities.VDS;
+import org.ovirt.engine.core.common.businessentities.VM;
+import org.ovirt.engine.core.common.businessentities.VdsCpuUnit;
+import org.ovirt.engine.core.common.utils.CpuPinningHelper;
+import org.ovirt.engine.core.compat.Guid;
+import org.ovirt.engine.core.vdsbroker.ResourceManager;
+
+public class VdsCpuUnitPinningHelper {
+    @Inject
+    private ResourceManager resourceManager;
+
+    /**
+     * This function will tell if the host is capable to run a given dedicated CPU policy VM.
+     *
+     * The function first will pin the CPU pending resources to the host, then:
+     * 1. If the VM number of vCPUs is smaller or equal to the pCPU exists on a single socket:
+     *    Pass on each socket and check if there are enough free pCPU to use.
+     *    If there are enough pCPUs - return true, otherwise false.
+     * 2. If the VM number of vCPUs is bigger to the pCPU exists on a single socket:
```

Comment:
phrasing:
1. drop the VM, as above
2. bigger to -> bigger than 
3. number of pCPUs on a single socket, as above